store/store.go

```diff
@@ -224,6 +224,12 @@ func (s *DefaultStore) LoadValidators(height uint64) (*tmtypes.ValidatorSet, err
 	if err != nil {
 		return nil, fmt.Errorf("failed to unmarshal to protobuf: %w", err)
 	}
+	if pbValSet.GetProposer() == nil {
+		return &tmtypes.ValidatorSet{
+			Validators: make([]*tmtypes.Validator, 0),
+			Proposer:   nil,
+		}, nil
+	}
```

Comment:
[question] Instead of checking the proposer, should we check that the validator set is empty?

Reply:
Sounds good, those checks should be equivalent but checking if the validator set is empty seems clearer. Thanks!

Reply:
Checking all 3 properties makes it even more explicit. Proposer = nil, Validator set = empty and totalVotingPower = 0

Reply:
In other words if Validator set = empty but not (Proposer = nil & totalVotingPower = 0) then it is an error. Checking this explicitly could be good.

Reply:
I think this falls into the category of over-checking. Notice that https://github.com/tendermint/tendermint/blob/64747b2b184184ecba4f4bffc54ffbcb47cfbcb0/types/validator_set.go#L326 already checks for the length of the validator set to be 0 and https://github.com/tendermint/tendermint/blob/64747b2b184184ecba4f4bffc54ffbcb47cfbcb0/types/validator_set.go#L317 already depends on the validator set.

In my opinion, adding unnecessary checks like these would add unneeded complexity to the code.